Quadrants

SimonFlapse opened the pull request:
Based on the map suggestion in #714

Four isolated quadrants must come together to launch a rocket

- [x] Resource transfer between quadrants
- [x] Prevent logistical and electrical network connection between quadrants
- [x] Teleportation between quadrants from spawn area (Only if empty inventory)
- [x] Identical spawn 
- [x] Oil patch for quadrant 3
- [x] Recipe restrictions on each quadrant (Currently broken in MP)
- [x] Research restricted to quadrant 1
- [x] 4 friendly forces
- [x] Increased ore count for quadrant 3
- [x] Restricted building capabilities on other quadrants
- [x] Automatic inventory empty to chest
- [x] Adapt recipes for 0.17 changes
- [x] Added train crossings
- [x] Added force_sync for chat and chart tags
- [x] Added informational compilatrons
- [x] Converted strings to localizedString
- [x] Added danish translation

Needs #795 

Closes #714

SimonFlapse committed:
Initial mapgen prototype

SimonFlapse committed:
Identical spawns, moved to combined

SimonFlapse committed:
Merge branch 'develop' of https://github.com/SimonFlapse/RedMew into Quadrants

SimonFlapse committed:
Switching teams added

SimonFlapse committed:
Technology and research restrictions added

SimonFlapse commented:
I now see that I actually deleted the test.lua map during the initial creation of this map. I can't see the usage of test.lua anyways, so unless we'd like it, I'd just leave it as deleted.

SimonFlapse committed:
Fixed multiplayer research

SimonFlapse committed:
Removed debugging logs

Forgot to remove writing to log

SimonFlapse committed:
Fixed new line at EoL

SimonFlapse committed:
Added oil for quadrant 3

SimonFlapse commented:
I hope to run a play test before 0.17 is out.

SimonFlapse committed:
Updating map info

Preparation for test

SimonFlapse committed:
Added chat information

SimonFlapse committed:
Increased resources for quadrant 2

As per suggestion in #714

SimonFlapse committed:
Fixed linting errors

SimonFlapse commented:
**Quadrant selection GUI:**
![image](https://user-images.githubusercontent.com/44922798/52947592-75ae6900-3377-11e9-9266-7908cf3766be.png)
**Map view:**
![image](https://user-images.githubusercontent.com/44922798/52947579-6c250100-3377-11e9-9e7d-b2ec430bd6c1.png)

SimonFlapse committed:
Minor tweaking of ores

SimonFlapse committed:
Restricted build capabilities on other teams land

SimonFlapse committed:
Fixed linting error

SimonFlapse committed:
Fixed players being stuck on teleport

grilledham commented on map_gen/maps/quadrants/scenario.lua:
```diff
@@ -0,0 +1,274 @@
+local start_oil = b.resource(oil_rectangle, 'crude-oil', b.exponential_value(100000, 0, 1))
+local start_tree_2 = b.entity(tree_rectangle_2, 'tree-01')
+
+start_iron = b.combine({ b.translate(start_iron, base_x, base_y), b.translate(start_iron, -base_x, -base_y), b.translate(start_iron, base_x, -base_y), b.translate(start_iron, -base_x, base_y) })
```
I think you want `b.any` rather than `b.combine`

SimonFlapse replied:
I'm actually not sure of the difference (For the guide) but yeah I'd any too now after having looked closer at the builder

SimonFlapse committed:
Item to chest function

Converted builder usage to using any instead of combine
Added item_to_chest that empties the players inventory to a chest if toggled on.

The chest is deleted if emptied by a player

SimonFlapse commented:
I've added the ability to empty your inventory into created chests.
Considerations:
- Locking it behind donator rank and having to buy it in the market
- Can spawn chests on tutorial tiles
- Can be used like a regular chest, giving the player an early steel chest (Cannot be mined)
- Might spawn in another quadrant if it cannot find a non colliding position

SimonFlapse committed:
Added improved hydra scaling

Added min, max and trigger to config
Added online_player_scale to config

Made hail hydra chances a variable of: - number of online players (toggleable)
- evolution based on a custom formula that has a slow scaling until evolution 0.8

SimonFlapse committed:
Added improved hydra scaling

Added min, max and trigger to config
Added online_player_scale to config

Made hail hydra chances a variable of: - number of online players (toggleable)
- evolution based on a custom formula that has a slow scaling until evolution 0.8

SimonFlapse committed:
Removed debug game.prints

SimonFlapse committed:
Removed debug game.prints

SimonFlapse committed:
Backwards compatibility

Added conversion from old config to new

SimonFlapse committed:
Backwards compatibility

Added conversion from old config to new

SimonFlapse committed:
Updated config to show -1 for unused

SimonFlapse committed:
Updated config to show -1 for unused

SimonFlapse committed:
Added locking chance

SimonFlapse committed:
Added locking chance

SimonFlapse committed:
Merge remote-tracking branch 'upstream/develop' into Quadrants

SimonFlapse committed:
Update for 0.17.1

SimonFlapse committed:
Added behemoth-worm-turret

SimonFlapse committed:
Merge remote-tracking branch 'origin/hydra-scaling' into hydra-scaling

# Conflicts:
#	config.lua
#	map_gen/shared/hail_hydra.lua

SimonFlapse committed:
Update for 0.17.4, Added Compilatron

SimonFlapse committed:
Merge remote-tracking branch 'upstream/develop' into Quadrants

SimonFlapse committed:
Fix ends -> end

SimonFlapse committed:
More text and removed game.print

SimonFlapse committed:
Hail hydra added, more help text

SimonFlapse committed:
Merge remote-tracking branch 'origin/hydra-scaling' into Quadrants

SimonFlapse committed:
Hail_hydra

SimonFlapse committed:
All_chat

SimonFlapse committed:
Chat_fix

SimonFlapse committed:
Colors and server protect

SimonFlapse committed:
Removed shout message

SimonFlapse committed:
Added player force to all_chat

SimonFlapse committed:
Updated version

SimonFlapse committed:
More messages

SimonFlapse committed:
Linting fix and tooltip added

SimonFlapse committed:
Revert "Merge remote-tracking branch 'origin/hydra-scaling' into Quadrants"

This reverts commit 80d56c2e4a00a58bd924a4cde904c8b3f08a99a9, reversing
changes made to 92841bc13ca5b0ecd2c0346540ad10b9361273ac.

SimonFlapse committed:
Fixes and minor additions

Player count
Where -> Were
How to Chat
Allowing pipes
Unminable chests explained by compilatron
Better too heavy message
Fixed auto shout formating

Initial code for Map tag sync

SimonFlapse committed:
Minor additions

Research message on research finished
Chart tags sync with all forces
- Needs refactoring
Quadrants labeled with a number
Evolution factors halved (except: Pollution from 150 -> 50)

Fixed linting

Added position to item_to_chest to override player position

SimonFlapse committed:
Added train_crossings

Allowing for trains to cross.
Handles the player switcing force and more.

Reverting pushed config.lua

SimonFlapse committed:
Fixed linting

SimonFlapse committed:
Fixed evolution factors

mheguy commented on config.lua:
```diff
@@ -1,5 +1,5 @@
-_DEBUG = false
-_CHEATS = false
+_DEBUG = true
+_CHEATS = true
```
You can (and should) set all your config options in `map_selection.lua` when you're testing. :D

mheguy commented on map_gen/maps/quadrants/switch_team.lua:
```diff
@@ -28,6 +28,8 @@ Global.register(
     end
 )
 
+local player_switched_force = script.generate_event_name()
```
I'm confused.. It looks like you created an event to call a local function?

SimonFlapse replied:
I guess I just wanted to try it out :P But in retro perspective it has no uses since I should use on_player_changed_force

SimonFlapse committed:
Fixed gui bug

mheguy commented on map_gen/maps/test.lua:
```diff
@@ -1,13 +0,0 @@
-local b = require 'map_gen.shared.builders'
```
Why are you deleting this?

SimonFlapse replied:
Started out as a mistake, because I used that file to make the map gen before I moved it into a folder. This caused a conflict an I ended up deleting it instead of discarding. I can bring it back if we need it.

mheguy replied:
Yes please. It's handy for what it is: a quick way to test worldgen.

SimonFlapse replied:
I use default for that, but that's not the right way I reckon

SimonFlapse committed:
Remove custom event

Increased teleport range
Limiting bridges to 12 chunks

SimonFlapse committed:
Reborn test.lua

SimonFlapse committed:
Forgot to remove this

mheguy commented on map_gen/maps/quadrants/switch_team.lua:
```diff
@@ -93,7 +93,6 @@ local function teleport(event, quadrant)
             nil,
             'Quadrants.quadrant_description'
         )
-        script.raise_event(player_switched_force, {extra = 'data'})
```
You could just run `update_gui(nil, true)` here, no?

SimonFlapse replied:
I already fixed that?

mheguy replied:
I should have been more clear:
Your solution was to create a function `changed_force` that calls another function `update_gui`. Why not just call `update_gui` directly?

SimonFlapse replied:
Well I'm not sure why I did it. I just had in mind that I wanted to try to use a custom event

mheguy committed:
Remove unused arg

SimonFlapse committed:
Added localization

mheguy commented on map_gen/maps/quadrants/force_sync.lua:
```diff
@@ -109,8 +109,7 @@ local function research_finished(event)
         if (string.find(force.name, 'quadrant')) ~= nil then
-            force.print(
-                '[img=item/automation-science-pack] New research complete: [technology=' .. technology.name .. ']'
+            force.print({'quadrants.force_sync_research', technology.name}
```
Rather than the tech name, you can take the tech’s localized name.

SimonFlapse replied:
I'm trying to display the technology sprite in chat, it'll convert it to localized string by itself when showing the tooltip for the technology

mheguy replied:
Ah ok, I get you. Cool.

mheguy reviewed (commented):
On phone, but looks good overall.

SimonFlapse committed:
Merge branch 'develop' into Quadrants
